src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMFeatureSelector.h

```diff
@@ -0,0 +1,265 @@
+      @param[out] result Transitions' names filtered out of the LP problem
+    */
+    virtual void optimize(
+      const std::vector<std::pair<double, String>>& time_to_name,
+      const std::map<String, std::vector<Feature>>& feature_name_map,
+      std::vector<String>& result
+    ) = 0;
+
+    /**
+      Splits the features into time segments. `optimize()` method is run on each of these segments.
+
+      @param[in] features Input features
+      @param[out] selected_filtered Output features
+    */
+    void select_MRMFeature(const FeatureMap& features, FeatureMap& selected_filtered);
+
+    void setNNThreshold(const Int nn_threshold);
+    Int getNNThreshold() const;
+
+    void setLocalityWeight(const bool locality_weight);
+    bool getLocalityWeight() const;
+
+    void setSelectTransitionGroup(const bool select_transition_group);
+    bool getSelectTransitionGroup() const;
+
+    void setSegmentWindowLength(const Int segment_window_length);
+    Int getSegmentWindowLength() const;
+
+    void setSegmentStepLength(const Int segment_step_length);
+    Int getSegmentStepLength() const;
+
+    void setVariableType(const String& variable_type);
+    String getVariableType() const;
+
+    void setOptimalThreshold(const double optimal_threshold);
+    double getOptimalThreshold() const;
+
+    void setScoreWeights(const std::map<String, String>& score_weights);
+    std::map<String, String> getScoreWeights() const;
+
+protected:
+    /**
+      Add variable to the LP problem instantiated in `optimize()`
+
+      @param[in,out] problem LPWrapper object
+      @param[in] name Column name
+      @param[in] bounded Double bounded if true, otherwise Unbounded.
+      @param[in] obj Objective value
+
+      @return The variable's column index
+    */
+    Int addVariable_(
+      LPWrapper& problem,
+      const String& name,
+      const bool bounded = true,
+      const double obj = 1.0
+    ) const;
+
+    /**
+      Scoring method used by the optimizer. Metavalues to use are decided by
+      the `score_weights_` class' member.
+      The returned value is used in the LP problems' variables and contraints.
+
+      @param[in] feature Input feature
+
+      @return Computed score
+    */
+    double computeScore_(const Feature& feature) const;
+
+    /**
+      Add constraint to the LP problem instantiated in `optimize()`
+
+      @param[in,out] problem LPWrapper object
+      @param[in] indices LP matrix indices
+      @param[in] values LP matrix values
+      @param[in] name Row name
+      @param[in] lb Lower bound
+      @param[in] ub Upper bound
+      @param[in] param Row type
+    */
+    void addConstraint_(
+      LPWrapper& problem,
+      std::vector<Int> indices,
+      std::vector<double> values,
+      const String& name,
+      const double lb,
+      const double ub,
+      const LPWrapper::Type param
+    ) const;
+
+private:
+    Int    nn_threshold_            = 4;
```

Comment:
these variables look familiar ;) how do they relate to SelectorParameters?

Reply:
I refactored the parameters. I moved `SelectorParameters` to `MRMFeatureSelector`. The class takes a single objects, and then its members are accessed by the derived classes.

I had to do this, otherwise the scheduler would not find the (moved) setters in the base class.